src/lando/api/legacy/workers/automation_worker.py

```diff
@@ -0,0 +1,189 @@
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.last_job_finished = None
+        self.refresh_active_repos()
+
+    def loop(self):
+        logger.debug(
+            f"{len(self.worker_instance.enabled_repos)} "
+            "enabled repos: {self.worker_instance.enabled_repos}"
+        )
+
+        # Refresh repos if there is a mismatch in active vs. enabled repos.
+        if len(self.active_repos) != len(self.enabled_repos):
+            self.refresh_active_repos()
+
+        if self.last_job_finished is False:
+            logger.info("Last job did not complete, sleeping.")
+            self.throttle(self.worker_instance.sleep_seconds)
+            self.refresh_active_repos()
+
+        with transaction.atomic():
+            job = AutomationJob.next_job(repositories=self.enabled_repos).first()
+
+        if job is None:
+            self.throttle(self.worker_instance.sleep_seconds)
+            return
+
+        with job.processing():
+            job.status = LandingJobStatus.IN_PROGRESS
+            job.attempts += 1
+            job.save()
+
+            # Make sure the status and attempt count are updated in the database
+            logger.info("Starting automation job", extra={"id": job.id})
+            self.last_job_finished = self.run_automation_job(job)
+            logger.info("Finished processing automation job", extra={"id": job.id})
+
+    def run_automation_job(self, job: AutomationJob) -> bool:
+        """Run an automation job."""
+        repo = job.target_repo
+        scm = repo.scm
+
+        with scm.for_push(job.requester_email):
+            repo_pull_info = f"tree: {repo.tree}, pull path: {repo.pull_path}"
+            try:
+                # TODO should we always update to the latest pull_path for a repo?
+                # or perhaps we need to specify some commit SHA?
+                scm.update_repo(repo.pull_path)
+            except SCMInternalServerError as e:
+                message = (
+                    f"`Temporary error ({e.__class__}) "
+                    f"encountered while pulling from {repo_pull_info}"
+                )
+                logger.exception(message)
+                job.transition_status(LandingJobAction.DEFER, message=message)
+
+                # Try again, this is a temporary failure.
+                return False
+            except Exception as e:
+                message = f"Unexpected error while fetching repo from {repo.pull_path}."
+                logger.exception(message)
+                job.transition_status(
+                    LandingJobAction.FAIL,
+                    message=message + f"\n{e}",
+                )
+                return True
+
+            # Run each action for the job.
+            actions = job.actions.all()
+            for action_row in actions:
+                # Turn the row action into a Pydantic action.
+                action = TypeAdapter(Action).validate_python(action_row.data)
```

Comment:
I'd suggest hiding this logic within the api logic, and just have something like a `ResolveAction(action_row.data)` here.